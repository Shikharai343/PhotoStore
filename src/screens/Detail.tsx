import React, { FC, useState } from 'react'
import { View, TouchableOpacity, StyleSheet } from 'react-native'
import { useRoute } from '@react-navigation/native'
import { DateUtil } from '../utils/DateUtil';
import IconComponent from '../components/IconComponent';
import ImageComponent from '../components/ImageComponent';
import Text from '../components/Text';
import { IParams } from '../navigation/interfaces';

const Detail: FC = () => {
    const { params } = useRoute();
    const [isFavourite, setIsFavourite] = useState(false)
    const param = params as IParams;

    if (!param || !param?.data) return <View />

    const onFavourite = () => {
        setIsFavourite(!isFavourite)
    }

    const { created_utc, url, title, author, total_awards_received } = param.data;
    const timestemp = DateUtil.timeDifference(created_utc);

    return (
        <View style={[styles.flexOne, styles.container]}>
            <View style={styles.content}>
                <View style={[styles.rowContainer, styles.details]}>
                    <Text
                        textType='label'
                        fontType='small'
                        color='#5B5E66'
                        style={styles.author}>
                        {`Posted by ${author} ${timestemp}`}
                    </Text>
                    <View style={styles.rowContainer}>
                        <IconComponent
                            name={"trophy"}
                            size={15}
                            color={'orange'}
                        />
                        <Text
                            textType='label'
                            fontType='small'
                            style={styles.award}>
                            {total_awards_received}
                        </Text>
                    </View>
                </View>
                <Text
                    textType='title'
                    fontType='small'
                    color='#43474E'
                    style={styles.title}>
                    {title}
                </Text>
            </View>
            <ImageComponent uri={url} imageStyle={styles.flexOne} containerStyle={styles.flexOne} />
            <TouchableOpacity activeOpacity={0.5} onPress={onFavourite} style={styles.favContainer}>
                <IconComponent name={"heart"} size={25} color={isFavourite ? 'red' : 'black'} />
            </TouchableOpacity>
        </View>
    )
}

export default Detail

const styles = StyleSheet.create({
    flexOne: {
        flex: 1,
    },
    container: {
        backgroundColor: 'white',
    },
    content: {
        padding: 16,
    },
    details: {
        marginVertical: 4,
    },
    author: {
        flex: 1,
        marginRight: 6,
    },
    award: {
        marginHorizontal: 4,
    },
    rowContainer: {
        flexDirection: 'row',
        justifyContent: "space-between",
    },
    title: {
        marginTop: 6
    },
    favContainer: {
        position: 'absolute',
        bottom: 50,
        ight: 50,
        alignSelf: 'flex-end',
        borderRadius: 25,
        padding: 10,
        backgroundColor: 'white',
    }

})