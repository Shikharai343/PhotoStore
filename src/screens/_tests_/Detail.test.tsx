import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import Detail from '../Detail';

jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon')
jest.mock('../../components/ImageComponent', () => 'ImageComponent')
jest.mock('@react-navigation/native', () => ({
    ...jest.requireActual('@react-navigation/native'),
    useRoute: () => ({
        params: {
            data: 'imageData',
        }
    }),
}));

describe('Detail Screen', () => {
    const wrapper: ShallowWrapper = shallow(
        <Detail />
    );

    it('should match snapshot with params', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })
});