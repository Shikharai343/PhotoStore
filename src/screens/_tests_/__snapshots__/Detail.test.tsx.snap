// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Screen should match snapshot with params 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "padding": 16,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          Object {
            "marginVertical": 4,
          },
        ]
      }
    >
      <Text
        color="#5B5E66"
        fontType="small"
        style={
          Object {
            "flex": 1,
            "marginRight": 6,
          }
        }
        textType="label"
      >
        Posted by undefined approximately NaN years ago
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <IconComponent
          color="orange"
          name="trophy"
          size={15}
        />
        <Text
          fontType="small"
          style={
            Object {
              "marginHorizontal": 4,
            }
          }
          textType="label"
        />
      </View>
    </View>
    <Text
      color="#43474E"
      fontType="small"
      style={
        Object {
          "marginTop": 6,
        }
      }
      textType="title"
    />
  </View>
  <ImageComponent
    containerStyle={
      Object {
        "flex": 1,
      }
    }
    imageStyle={
      Object {
        "flex": 1,
      }
    }
  />
  <TouchableOpacity
    activeOpacity={0.5}
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "white",
        "borderRadius": 25,
        "bottom": 50,
        "ight": 50,
        "padding": 10,
        "position": "absolute",
      }
    }
  >
    <IconComponent
      color="black"
      name="heart"
      size={25}
    />
  </TouchableOpacity>
</View>
`;
